src/triage/component/architect/feature_generators.py

```diff
@@ -560,6 +563,26 @@ def index_column_lookup(self, aggregations, imputed=True):
             for aggregation in aggregations
         )
 
+    def _needs_features(self, aggregation):
+        imputed_table = self._clean_table_name(
+            aggregation.get_table_name(imputed=True)
+        )
+
+        if self._table_exists(imputed_table):
+            check_query = f"select 1 from {aggregation.state_table} " +\
+                f"left join {self.features_schema_name}.{imputed_table} using (entity_id, as_of_date) " +\
+                f"where {self.features_schema_name}.{imputed_table}.entity_id is null limit 1"
```

Comment:
I think the addition operators are unnecessary. Otherwise, parentheses around the entire expression might be nicer than line break escapes.